cylc/sphinx_ext/cylc_lang/autodocumenters.py

```diff
@@ -157,82 +232,98 @@ class CylcAutoDirective(Directive):
-                # section
-                ret.extend(
-                    cls.doc_section(key)
-                )
-                section_partials = partials + [('section', key)]
-                ret.extend([
-                    f'   {line}'
-                    for line in cls.doc_spec(value, section_partials)
-                ])
-            elif isinstance(value, list):
-                # setting
-                ret.extend(
-                    cls.doc_setting(key, *value)
-                )
-            else:
-                raise TypeError(value)
-        return ret
-
     def run(self):
-        conf_name = self.arguments[0].strip()
-        if len(self.arguments) == 2:
-            spec = get_obj_from_module(self.arguments[1].strip())
+        if len(self.arguments) == 1:
+            spec = get_obj_from_module(self.arguments[0].strip())
         else:
             spec = json.loads('\n'.join(self.content))
 
-        content = []
+        content = doc_spec(spec)
+
+        # parse the RST text
+        node = addnodes.desc_content()
+        self.state.nested_parse(
+            StringList(content),
+            self.content_offset,
+            node
+        )
 
-        # document the top-level configuration
+        return [node]
+
+
+def doc_type(typ):
+    content = []
+    if 'help' in typ:
+        content.append(dedent(typ['help']).strip())
+        content.append('')
+    if 'examples' in typ:
+        examples = typ['examples']
+        content.extend(
+            directive('rubric', ['Examples:'])
+        )
+        if isinstance(examples, list):
+            examples = {k: None for k in examples}
+        for example, notes in examples.items():
+            content.append(
+                f'* ``{example}``'
+                + (f' - {notes}' if notes else '')
+            )
+        content.append('')
+    if 'references' in typ:
         content.extend(
-            self.doc_conf(conf_name)
+            directive('rubric', ['See Also:'])
         )
+        for role, string in typ['references']:
+            content.append(
+                f'* :{role}:`{string}`'
+            )
+        content.append('')
 
-        # document the SPEC
-        content.extend([
-            f'   {line}'
-            for line in self.doc_spec(spec, [('conf', conf_name)])
-        ])
+    return directive(
+        'parsec:type',
+        [typ['name']],
+        {},
+        {},
+        '\n'.join(content)
+    )
+
+
+class CylcAutoTypeDirective(Directive):
+    """Auto-documenter for Parsec configuration types (i.e. validators)."""
+
+    INFO_FIELDS = [
+        'name',
+        'help',
+        'examples',
+        'references'
+    ]
+
+    has_content = True
+    option_spec = {}
+    required_arguments = 0
+    optional_arguments = 9
+
+    @classmethod
+    def iter_types(cls, objects):
+        types = {}
+        for obj in objects:
+            types.update(obj)
+        for name, info in sorted(types.items()):
```

Comment:
```suggestion
        for _, info in sorted(types.items()):
```
My pylint is complaining about unused varible name  - pycodestyle seems happy though! Happy to ignore.